src/alto-ui/Form/TextField/TextField.js

```diff
@@ -38,54 +47,76 @@ const TextField = React.forwardRef((props, ref) => {
     error,
     area,
     right,
-    ghost,
-    frozen,
     disabled,
     percent,
+    children,
+    clearable,
+    onClear,
     ...remainingProps
   } = props;
+  const defaultRef = useRef();
+  const ref = passedRef || defaultRef;
+  const visibilityProps = props.readOnly ? { 'aria-hidden': true, tabIndex: '-1' } : {};
 
-  const visibilityProps = frozen ? { 'aria-hidden': true, tabIndex: '-1' } : {};
-
-  if (ghost) return <GhostInput {...props} />;
-
-  const className = bemClass(
-    'textfield',
-    {
-      large,
-      small,
-      success,
-      error,
-      area,
-      right,
-      frozen,
-    },
-    props.className
-  );
+  const [isFocus, setFocus] = useState(false);
+
+  const modifiers = {
+    large,
+    small,
+    success,
+    error,
+    area,
+    right,
+    focus: isFocus,
+    disabled: props.disabled,
+    readOnly: props.readOnly,
+  };
 
-  return (
-    <FormElement {...props}>
-      {area ? (
-        <textarea
-          ref={ref}
-          {...remainingProps}
-          className={className}
-          {...visibilityProps}
-          disabled={disabled || frozen}
-        />
-      ) : (
-        <input
-          ref={ref}
-          {...remainingProps}
-          className={className}
-          type={getInputType(props.type)}
-          disabled={disabled || frozen}
-          {...visibilityProps}
-          onChange={event => handleChange(event, props.type, props.onChange)}
-        />
+  const className = bemClass('textfield', modifiers, props.className);
+  const inputClassName = bemClass('textfield__input', modifiers);
+  const renderInput = typeof children === 'function' ? children : x => x;
+
+  const element = area ? (
+    <textarea
+      ref={ref}
+      {...remainingProps}
+      className={className}
+      {...visibilityProps}
+      disabled={disabled}
+    />
+  ) : (
+    <div className={className}>
+      {renderInput(
+        <Fragment>
```

Comment:
```suggestion
        <>
```

Reply:
There is missing configuration on 8.3 to enable shortcut for Fragment